### DISCUSSION TITLE
Crash STTextView/NSTextContentManager+Helpers.swift:55: Precondition failed + lookup shows wrong word

### DISCUSSION THREAD

**Discussion by tarasis** [krzyzanowskim/STTextView#21] [Questions]
Was recording a video showing that the lookup action (three finger tap) wasn't behaving correctly in the TextEdit sample app, when I clicked on another word and the app crashed and opened debugger in STTextView/NSTextContentManager+Helpers.swift:55: Precondition failed

macOS 13.4, Xcode 14.3.1, STTextView 0.7.2

(Vid 1 I had the box for screen record too small and so it chops part of the definitions off, but its enough you can still see its the wrong word & definition)
https://github.com/krzyzanowskim/STTextView/assets/3006/5c2a025f-2286-4ca7-8c69-1e69bf875932

Another video, showing to close to beginning and it won't show any definition for 3 finger tap. Further down it shows for a word above where I tapped and then the crash.

https://youtu.be/MrreVUXcZDs (video was 30mb, max size for here is 10mb, apologies)

**Comment by krzyzanowskim** [comment]
thank you! looks like it hit the precondition that "should not happen"

<img width="711" alt="Screenshot 2023-07-09 at 17 39 45" src="https://github.com/krzyzanowskim/STTextView/assets/758033/f60a2fb5-47e0-4dbc-8741-9cb2e87a0302">

I can easily reproduce it. I don't have this action enabled on my side, that's why it never occurred to me ;)

**Reply by tarasis** [reply]
Thats entirely fair, I use it a bunch* but I had accidentally triggered it which is why I noticed.

* English UK, English US, German to English and Wikipedia

**Comment by krzyzanowskim** [comment]
fixed on `main` branch. please confirm

**Comment by tarasis** [comment]
Thank you! Can confirm that definitions now correctly show for the word tapped! Woot!

However the crash, because of the precondition, can/does still occur if you accidentally do the three finger tap on a space / newline between the paragraphs. (For instance three finger tap on line 2) But it only seems to do it in that circumstance, and I tried to crash it :)

Aside: three finger tap on the annotation generates the log

```
2023-07-09 21:00:41.566262+0100 Text Editor[14003:6585439] Lookup: Unhandled exception 'NSInvalidArgumentException' caught in +[LULookupDefinitionModule _focusTermUsingQueue:handler:]
```

**Comment by krzyzanowskim** [comment]
right! I didn't test that. It should be handled now.

**Reply by tarasis** [reply]
Perfect ✅
